src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -43,26 +47,149 @@ struct segment_info_t {
   }
 };
 
+// position within segment_info_set_t
+using segment_pos_t = uint64_t;
+
+struct segment_manager_info_t {
+  device_id_t device_id;
+  segment_id_t num_segments;
+  segment_pos_t segment_offset;
+};
+
+class segment_info_set_t {
+public:
+  segment_info_t& operator[](device_segment_t id) {
+    auto d_id = id.device_id();
+    auto& segment_manager = segment_managers[d_id];
```

Comment:
auto &sm_info -- it doesn't have type SegmentManager& so you probably should call it segment_manager.